core/templates/domain/feedback_thread/FeedbackThreadObjectFactory.ts

```diff
@@ -46,40 +50,58 @@ export class FeedbackThread {
     this.messageCount = messageCount;
     this.stateName = stateName;
     this.threadId = threadId;
-    this.messages = [];
+    this.lastNonemptyMessageSummary = lastNonemptyMessageSummary;
   }
 
-  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
-  // 'any' because 'messages' is an array of dicts with underscore_cased keys
-  // which give tslint errors against underscore_casing in favor of camelCasing.
-  setMessages(messages: any[]): void {
+  setMessages(messages: ThreadMessage[]): void {
     this.messages = messages;
+    // Since messages have been updated, we need to update all of our other
+    // message-related fields to maintain consistency between them.
+    this.messageCount = messages.length;
+    let nonemptyMessages = messages.filter(m => m.isNonempty());
+    if (nonemptyMessages.length > 0) {
+      let i = nonemptyMessages.length - 1;
+      this.lastNonemptyMessageSummary = nonemptyMessages[i].summary;
+    }
+  }
+
+  getMessages(): ThreadMessage[] {
+    return this.messages;
   }
 
   isSuggestionThread(): boolean {
     return false;
   }
 }
 
-@Injectable({
-  providedIn: 'root'
-})
+@Injectable({providedIn: 'root'})
 export class FeedbackThreadObjectFactory {
+  constructor(
+    private threadMessageSummaryObjectFactory:
+      ThreadMessageSummaryObjectFactory) {}
+
   // TODO(#7176): Replace 'any' with the exact type. This has been kept as
   // 'any' because 'feedbackThreadBackendDict' is a dict with underscore_cased
   // keys which give tslint errors against underscore_casing in favor of
   // camelCasing.
   createFromBackendDict(feedbackThreadBackendDict: any): FeedbackThread {
+    let lastNonemptyMessageSummaryBackendDict = {
```

Comment:
How about creating the domain object here, so that we aren't defining variables that are raw dicts. 

It also seems a bit weird to construct a backend dict in the frontend like this. Perhaps better to use another method: threadMessageSummaryObjectFactory.createNew(text, authorUsername)